contrib/wireshark/rmpp.lua

```diff
@@ -0,0 +1,271 @@
+--
+-- Copyright (c) NVIDIA CORPORATION & AFFILIATES, 2023. ALL RIGHTS RESERVED.
+-- See file LICENSE for terms.
+--
+
+-- Usage:
+-- $ wireshark -Xlua_script:rmpp.lua <pcap_file>
+--
+-- or copy the LUA file in the Wireshark plugins directory.
+--   - Help -> About Wireshark -> Folders -> Personal Lua plugins
+
+local rmpp_version_str = {
+    [0] = "Not Set",
+    [1] = "1",
+}
+
+local RMPP_STATUS_SUCCESS    =   0
+local RMPP_STATUS_RESX       =   1
+local RMPP_STATUS_ABORT_MIN  = 118
+local RMPP_STATUS_T2L        = 118
+local RMPP_STATUS_BAD_LEN    = 119
+local RMPP_STATUS_BAD_SEG    = 120
+local RMPP_STATUS_BADT       = 121
+local RMPP_STATUS_W2S        = 122
+local RMPP_STATUS_S2B        = 123
+local RMPP_STATUS_BAD_STATUS = 124
+local RMPP_STATUS_UNV        = 125
+local RMPP_STATUS_TMR        = 126
+local RMPP_STATUS_UNSPEC     = 127
+
+local rmpp_status_str = {
+    [RMPP_STATUS_SUCCESS]    = "Success",
+    [RMPP_STATUS_RESX]       = "Resx",
+    [RMPP_STATUS_ABORT_MIN]  = "Abort Min",
+    [RMPP_STATUS_T2L]        = "T2L",
+    [RMPP_STATUS_BAD_LEN]    = "Bad Len",
+    [RMPP_STATUS_BAD_SEG]    = "Bad Seq",
+    [RMPP_STATUS_BADT]       = "BadT",
+    [RMPP_STATUS_W2S]        = "W2S",
+    [RMPP_STATUS_S2B]        = "S2B",
+    [RMPP_STATUS_BAD_STATUS] = "Bad Status",
+    [RMPP_STATUS_UNV]        = "Unv",
+    [RMPP_STATUS_TMR]        = "TMR",
+    [RMPP_STATUS_UNSPEC]     = "Unspec",
+}
+
+local RMPP_TYPE_DATA  = 1
+local RMPP_TYPE_ACK   = 2
+local RMPP_TYPE_STOP  = 3
+local RMPP_TYPE_ABORT = 4
+
+local rmpp_valid_type = {
+    [RMPP_TYPE_DATA]  = 1,
+    [RMPP_TYPE_ACK]   = 1,
+    [RMPP_TYPE_STOP]  = 1,
+    [RMPP_TYPE_ABORT] = 1,
+}
+
+local rmpp_type_str = {
+    [RMPP_TYPE_DATA]  = "Data",
+    [RMPP_TYPE_ACK]   = "Ack",
+    [RMPP_TYPE_STOP]  = "Stop",
+    [RMPP_TYPE_ABORT] = "Abort",
+}
+
+local RMPP_FLAG_ACTIVE = 1
+local RMPP_FLAG_FIRST  = 2
+local RMPP_FLAG_LAST   = 4
+
+local rmpp_flag_str = {
+    [RMPP_FLAG_ACTIVE] = "Active",
+    [RMPP_FLAG_FIRST]  = "First",
+    [RMPP_FLAG_LAST]   = "Last",
+}
+
+local rmpp_flag_set = {
+    [0] = "Not Set",
+    [1] = "Set",
+}
+
+local rmpp_proto = Proto("RMPP", "Reliable Message Passing Protocol")
```

Comment:
isn't it `Reliable  Multi-Packet Transaction Protocol`

Reply:
fixed, seems that there is another rmpp that has some similarities while not being directly related to infiniband.